### PR TITLE
Added a protocol readme.

### PR DESCRIPTION
**Pull request by knarfi**
The protocol is based on the existing protocol but converted to markdown and translated to english. The unsubscribe command is added.

### CODE REVIEW

**Comment by knarfi** [comment]
I do not fully understand why Travis creates errors or warnings on this, did I make daddy mad? If there is something to blame on my .md file, let me know.

**Comment by goto-bus-stop** [comment]
I'm +1 on a Markdown version.

If we add a Markdown version, we can remove the `.txt` file, so we'll only have to change one document if the protocol is changed.

`protocol.md` should go in the `doc/` folder.

(Travis is mad because it isn't configured, yet :') )

**Comment by Bennyn92** [comment]
+1. Looks fine

File: protocol.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+Items between brackets ('[' and ']') represent a list.
+Items between braces ('{' and '}') represent a map. As with all maps, the order has not been determined.
+
+###### Notes for client commands:
+The commands and arguments are not case sensitive. With exception from the names of players and gametypes
+
+**Not supported command:**  
+C: <not supported command>  
+S: ERR <reason>  
+->no action.
```
Comment (goto-bus-stop) [inline protocol.md]: The communication examples could also be code blocks.

---

**Unsupported command:**
```
C: <unsupported command>
S: ERR <reason>
```
->no action.

**Comment by imichaelv** [comment]
Nice, but SERVER_COMMANDS / client_commands (stick it to all caps or no caps) even though this could be ignored by code

**Comment by knarfi** [comment]
I just translated the original file and re-styled it. Of course any of you can edit it and more important, new commands need to be added as they are created. For now we should delete the .txt version and anyone who has time for small improvements should make them.
Since the server is case sensitive (if we can believe the original protocol) I do not know if I can just change the commands to all capital.

File: protocol.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+### Commands overview
+##### Overview of server-messages:
+
+```
+OK          Command accepted
+ERR         Command denied
+SVR [HELP | GAME [MATCH | YOURTURN | MOVE | CHALLENGE | [WIN | LOSS | DRAW]]]
+            Message from server
+    HELP            Message with help information
+    GAME            Message concerning the game
```
Comment (imichaelv) [inline protocol.md]: add 
`    GAMELIST       ["<gametype>",..] `
`    PLAYERLIST    ["<player>",..]`

change 
`SVR [HELP | GAME [MATCH | YOURTURN | MOVE | CHALLENGE | [WIN | LOSS | DRAW]]]`
to
`SVR [HELP | GAME [MATCH {GAMETYPE, PLAYERTOMOVE, OPPONENT} | YOURTURN | MOVE | CHALLENGE | [WIN | LOSS | DRAW]] | GAMELIST | PLAYERLIST]`

and 
 `      MATCH            Assigning a match with {GAMETYPE, PLAYERTOMOVE, OPPONENT}`

**Comment by goto-bus-stop** [comment]
> For now we should delete the .txt version and anyone who has time for small improvements should make them.

yez. please move protocol.md into the docs folder and delete the docs/protocol.txt file, then this is ready to go

**Comment by Jarinus** [comment]
Pull a current version of the branch in your local branch and push again. The Travis error says there is no Rake file and I'm pretty sure that's because you have a version which does not have the correct Travis configuration.

**Comment by knarfi** [comment]
I implemented the feedback and did some minor style changes. Also placing this comment to notify everyone via email :#

**Comment by IcyPalm** [comment]
Seems good to me! Merging

**Pull request merged**